shared_model/builders/common_objects/amount_builder.hpp

```diff
@@ -52,6 +54,45 @@ namespace shared_model {
         return copy;
       }
 
+      /**
+       * Constructs Amount object from given string
+       * @param str_amount string in format "100.00",
+       * where there are exactly precision numbers after dot
+       * @return Amount constructed from string
+       */
+      static BuilderResult<shared_model::interface::Amount> fromString(std::string str_amount) {
+        // taken from iroha::model::Amount
+        // check if valid number
+        std::regex e("([0-9]*\\.[0-9]+|[0-9]+)");
+        if (!std::regex_match(str_amount, e)) {
```

Comment:
Why it is here, but not in validator?

Reply:
Because this function parses a string into a number. Therefore it must check that string has the correct format. The validator checks if number is valid in a given context, and has nothing to do with strings